src/io/buf_read/mod.rs

```diff
@@ -41,6 +41,15 @@ cfg_if! {
 /// [`futures::io::AsyncBufRead`]:
 /// https://docs.rs/futures-preview/0.3.0-alpha.17/futures/io/trait.AsyncBufRead.html
 pub trait BufRead {
+    /// Tells this buffer that `amt` bytes have been consumed from the buffer, so they should no
+    /// longer be returned in calls to `read`.
+    fn consume(&mut self, amt: usize)
+    where
+        Self: AsyncBufRead + Unpin,
+    {
+        AsyncBufRead::consume(Pin::new(self), amt)
+    }
```

Comment:
Can we require only `Unpin` as a trait bound and not provide a default method implementation here? Instead, let's move the body of this method into the `impl<T: AsyncBufRead + Unpin + ?Sized> BufRead for T` block.

The reason is that `std::io::BufRead` does not provide default implementations for `fill_buf` and `consume` but does for other methods, so we should follow suit. :)